Assets/MixedRealityToolkit-SDK/Features/SpatialAwareness/System/MixedRealitySpatialAwarenessSystem.cs

```diff
@@ -157,68 +163,64 @@ public override void Destroy()
 
-        /// <inheritdoc />
         /// <summary>
-        /// The spatial awareness system will call the <see cref="IMixedRealitySpatialAwarenessMeshHandler.OnMeshAdded"/> method to indicate a new mesh has been added.
+        /// <summary>
+        /// Method called by the surface observer to inform the spatial awareness system that an existing mesh has been updated.
         /// </summary>
-        /// <param name="meshId">Value identifying the mesh.</param>
-        /// <param name="position">The position, in the environment, where the mesh should be placed.</param>
-        /// <param name="meshData">The mesh data.</param>
-        /// <param name="meshObject">The mesh subsystem managed <see cref="GameObject"/> for the mesh.</param>
-        private void RaiseMeshAdded(uint meshId, Vector3 position, Mesh meshData, GameObject meshObject = null)
+        /// <param name="meshId">The id of the mesh</param>
+        /// <param name="meshObject">The mesh <see cref="GameObject"/></param>
+        public void UpdateMesh(uint meshId, GameObject meshObject)
         {
-            // todo
-        }
+            // todo: update the mesh in the collection
 
-        /// <summary>
-        /// Event sent whenever a mesh is added.
-        /// </summary>
-        private static readonly ExecuteEvents.EventFunction<IMixedRealitySpatialAwarenessMeshHandler> OnMeshAdded =
-            delegate (IMixedRealitySpatialAwarenessMeshHandler handler, BaseEventData eventData)
-            {
-                // todo
-                //BoundaryEventData boundaryEventData = ExecuteEvents.ValidateEventData<BoundaryEventData>(eventData);
-                //handler.OnBoundaryVisualizationChanged(boundaryEventData);
-            };
+            RaiseMeshAvailable(meshId, meshObject);
+        }
 
-        /// <inheritdoc />
         /// <summary>
-        /// The spatial awareness system will call the <see cref="IMixedRealitySpatialAwarenessMeshHandler.OnMeshUpdated"/> method to indicate an exising mesh has changed.
+        /// The spatial awareness system will call the <see cref="IMixedRealitySpatialAwarenessMeshHandler.OnMeshUpdated"/> method to indicate a mesh has been added or updated.
         /// </summary>
         /// <param name="meshId">Value identifying the mesh.</param>
-        /// <param name="position">The position, in the environment, where the mesh should be placed.</param>
-        /// <param name="meshData">The mesh data.</param>
-        /// <param name="meshObject">The mesh subsystem managed <see cref="GameObject"/> for the mesh.</param>
-        private void RaiseMeshUpdated(uint meshId, Vector3 position, Mesh meshData, GameObject meshObject = null)
+        /// <param name="meshObject">The mesh <see cref="GameObject"/>.</param>
+        private void RaiseMeshAvailable(uint meshId, GameObject meshObject)
         {
             // todo
         }
 
         /// <summary>
         /// Event sent whenever a mesh is updated.
```

Comment:
This comment is likely incorrect. It seems like it would be when the mesh is made available, nut updated.